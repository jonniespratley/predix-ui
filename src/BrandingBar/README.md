The `<BrandingBar/>` component provides a header area to contain title, logo and branding content.


## Usage

```react
const {BrandingBar} = PxReact;
<BrandingBar title="Predix Design System"/>
```

## Properties

```table
span: 6
rows:
  - Name: hideLogo
    Type: Bool
    Description: To show the logo or not.
  - Name: hidePowered
    Type: Bool
    Description: To show the powered by or not.
  - Name: responsiveWidth
    Type: String
    Description: The window size in which to hide.
  - Name: title
    Type: String
    Description: The application title to display in the lefthand corner of the branding bar.
```


### Styling

```table
span: 6
rows: 
  - Custom Property: --px-branding-bar-height
    Description: Height of the branding bar
  - Custom Property: --px-branding-bar-background-color
    Description: Background color for the branding bar
  - Custom Property: --px-branding-bar-font-size
    Description: Font size for text displayed in the branding bar
  - Custom Property: --px-branding-bar-logo-and-title-text-color
    Description: Text color of the logo and title area of the branding bar
```