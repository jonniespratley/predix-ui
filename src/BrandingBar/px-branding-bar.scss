// Generic
//@import "px-normalize-design/_generic.normalize-mini.scss";
//@import "px-box-sizing-design/_generic.box-sizing.scss";
//@import "px-headings-design/_base.headings.scss";
@import "px-helpers-design/_generic.helpers.scss";
// Base
//@import "px-flexbox-design/_base.flexbox.scss";
//$inuit-enable-margins : true;
//@import 'px-spacing-design/_trumps.spacing.scss';
//$inuit-enable-responsive-margins : true;
//@import "px-spacing-responsive-design/_trumps.spacing-responsive.scss";
// Component
.px-branding-bar {
  
  @include media-query(palm) {
   
  }

}