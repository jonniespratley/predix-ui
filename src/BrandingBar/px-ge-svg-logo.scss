//@import "px-normalize-design/_generic.normalize-mini.scss";
