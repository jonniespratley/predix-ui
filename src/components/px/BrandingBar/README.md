The `<BrandingBar/>` component provides a header area to contain title, logo and branding content.

```hint
Reference https://www.predix-ui.com/#/elements/px-branding-bar
```


## Usage

```react
const {BrandingBar} = PxReact;
<BrandingBar
  application-title="Predix Design System"/>
```

## Properties

```table
span: 6
rows:
  - Name: applicationTitle
    Type: String
    Description: The application title to display in the lefthand corner of the branding bar.
```
