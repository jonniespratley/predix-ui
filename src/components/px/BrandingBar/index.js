import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import PredixLogo from './px-predix-svg-logo';
import Logo from './px-ge-svg-logo';
import Flex from '../../../styles/flex';

const BrandingBar = styled.div`
  display         : flex;
  align-items     : center;
  width           : inherit;
  justify-content : space-between;
  height          : var(--px-branding-bar-height, 30px);
  max-height      : 67px;
  min-height      : 30px;
  background-color: var(--px-branding-bar-background-color, black);
  font-size       : var(--px-branding-bar-font-size, 15px);
  color           : var(--px-branding-bar-logo-and-title-text-color, gray);
  padding         : 0 1rem;

  ${props => props.responsiveWidth && css`
      @media (max-width: ${props.responsiveWidth}) {
        display   : none !important;
        visibility: hidden;
    }
  `}
`;
BrandingBar.displayName = 'BrandingBar';
BrandingBar.defaultProps = {
  responsiveWidth: '768px'
};

const BrandingBarTitle = styled.div`
  margin-left: .5rem;
`;
BrandingBarTitle.displayName = 'BrandingBarTitle';

const BrandingBarPoweredBy = styled.div`
  font-size: 12px;
  margin-right: .3rem;
`;
BrandingBarPoweredBy.displayName = 'BrandingBarPoweredBy';


const Component = ({
  title,
  powered,
  hideLogo,
  customLogo,
  hidePowered,
  responsiveWidth,
  children
}) => (
  <BrandingBar responsiveWidth={responsiveWidth}>
    <Flex middle flex={1}>
      <Flex middle>
        {!customLogo && !hideLogo && <Logo />}
        {customLogo && customLogo()}
      </Flex>
      <Flex middle>
        <BrandingBarTitle>{title}</BrandingBarTitle>
      </Flex>
    </Flex>
    {children && <div>{children}</div>}

    {!hidePowered &&
    <Flex middle>
      <BrandingBarPoweredBy>{powered}</BrandingBarPoweredBy>
      <PredixLogo size={10} />
    </Flex>
  }
  </BrandingBar>
);

Component.propTypes = {
  title: PropTypes.string,
  powered: PropTypes.string,
  responsiveWidth: PropTypes.string,
  hideLogo: PropTypes.bool,
  hidePowered: PropTypes.bool,
  customLogo: PropTypes.func,
  children: PropTypes.node
};

Component.defaultProps = {
  title: 'App Name',
  responsiveWidth: '500px',
  powered: 'Powered by React',
  hideLogo: false,
  hidePowered: false,
  customLogo: null,
  children: null
};

Component.displayName = 'BrandingBarComponent';

export default Component;
