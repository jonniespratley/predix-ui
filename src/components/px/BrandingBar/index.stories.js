import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean } from '@storybook/addon-knobs/react';
import { withReadme } from 'storybook-readme';

import BrandingBar from './';
import Icon from '../IconSet/Icon';
import Button from '../Button';
import Flex from '../../../styles/flex';
import README from './README.md';

const customLogo = () => (
  <Icon icon="px-nav:home" />
);
const btnStyles = {
  color: 'white'
};
// / BrandingBar
storiesOf('Components / Branding Bar', module)
  .addDecorator(withKnobs)
  .addDecorator(withReadme(README))
  .addWithJSX('default', () => (
    <BrandingBar
      hideLogo={boolean('hideLogo', false)}
      hidePowered={boolean('hidePowered', false)}
      powered={text('powered', 'Powered by React')}
      responsiveWidth={text('responsiveWidth', '500px')}
      title={text('title', 'Application Title')}
    />
  ))
  .addWithJSX('without logo', () => (
    <BrandingBar title="Custom Title" hideLogo />
  ))

  .addWithJSX('without powered by', () => (
    <BrandingBar title="Custom Title" hidePowered />
  ))
  .addWithJSX('with custom logo', () => (
    <BrandingBar title="Custom Logo" customLogo={customLogo} hidePowered />
  ))
  .addWithJSX('with custom children', () => (
    <BrandingBar
      title="Platform"
      hidePowered
    >
      <Flex right className="app-user-menu">
        <Button theme="bare" style={btnStyles}>
          <Icon icon="px-nav:notification" />
        </Button>
        <Button theme="bare" style={btnStyles}>
          <Icon icon="px-nav:generic-user" />
        </Button>
      </Flex>
    </BrandingBar>
  ));

