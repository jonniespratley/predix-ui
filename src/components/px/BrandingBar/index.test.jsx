import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import BrandingBar from './';

describe('BrandingBar', () => {
  test('should render', () =>{
    const tree = renderer.create(<BrandingBar title="Custom Title" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should render with hideLogo', () =>{
    const tree = renderer.create(<BrandingBar title="Custom Title" hideLogo/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('should render with hidePowered', () =>{
    const tree = renderer.create(<BrandingBar title="Custom Title" hidePowered/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
