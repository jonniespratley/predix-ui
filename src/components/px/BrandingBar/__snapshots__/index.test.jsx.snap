// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandingBar should render 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: inherit;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: var(--px-branding-bar-height,30px);
  max-height: 67px;
  min-height: 30px;
  background-color: var(--px-branding-bar-background-color,black);
  font-size: var(--px-branding-bar-font-size,15px);
  color: var(--px-branding-bar-logo-and-title-text-color,gray);
  padding: 0 1rem;
}

.c3 {
  margin-left: .5rem;
}

.c4 {
  font-size: 12px;
  margin-right: .3rem;
}

@media (max-width:500px) {
  .c0 {
    display: none !important;
    visibility: hidden;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <svg
        height="20"
        viewBox="-280 372 50 50"
        width="20"
        x="0"
        y="0"
      >
        <path
          className="px-ge-svg-logo"
          d="M-255 422c-13.8 0-25-11.1-25-25 0-13.8 11.2-25 25-25s25 11.2 25 25C-230 410.9-241.2 422-255 422zM-255 373.2c-13.1 0-23.8 10.7-23.8 23.8s10.7 23.8 23.8 23.8 23.8-10.7 23.8-23.8S-241.8 373.2-255 373.2zM-233.8 404.1c0 0 0 0-0.1 0s-0.1 0-0.1-0.1c0 0 0.8-2.5 0.8-5.3 0-3-1.3-4.9-2.8-4.9 -1 0-1.7 0.7-1.7 1.7 0 1.9 2.3 2 2.3 6.1 0 1.7-0.4 3.3-0.9 5 -2.6 8.7-10.7 12.7-18.7 12.7 -3.7 0-6.3-0.8-7.1-1.1v-0.1l0.1-0.1c0.3 0.1 2.5 0.8 5.3 0.8 3 0 4.8-1.3 4.8-2.8 0-1-0.8-1.7-1.7-1.7 -1.9 0-2 2.4-6 2.4 -1.7 0-3.3-0.4-5.1-0.9 -8.6-2.6-12.7-10.7-12.7-18.8 0-3.9 1.1-7 1.1-7.1h0.1c0.1 0 0.1 0.1 0.1 0.1 -0.1 0.3-0.8 2.5-0.8 5.3 0 3 1.3 4.8 2.8 4.8 0.9 0 1.7-0.7 1.7-1.7 0-1.9-2.3-2.1-2.3-6.1 0-1.7 0.4-3.3 0.9-5.1 2.6-8.6 10.7-12.6 18.7-12.7 3.7 0 6.9 1.1 7.1 1.2v0.1c0 0.1-0.1 0.1-0.1 0.1s-2.1-0.9-5.3-0.9c-3 0-4.8 1.3-4.8 2.8 0 0.9 0.7 1.7 1.7 1.7 1.9 0 2-2.3 6-2.3 1.7 0 3.3 0.4 5.1 0.9 8.7 2.6 12.6 10.8 12.7 18.7C-232.6 401-233.8 404.1-233.8 404.1zM-245.7 398.2c-2.4 0-4.2 1.7-4.2 3.8 0 1.7 1 3.1 2.4 3.1 0.5 0 1-0.3 1-0.9 0-0.9-1.2-1.1-1.1-2.5 0-0.9 0.9-1.5 1.7-1.5 1.7 0 2.4 1.6 2.4 3.3 -0.1 2.6-2 4.3-4.2 4.3 -2.9 0-4.8-2.8-4.8-5.8 0-4.4 2.9-6.2 4.4-6.6 0 0 4 0.7 3.8-1 0-0.8-1.2-1.1-2-1.1 -0.9 0-1.8 0.3-1.8 0.3 -0.5-0.2-0.8-0.7-1-1.3 2.8-2.1 4.8-4.2 4.8-6.6 0-1.3-0.8-2.4-2.4-2.4 -2.8 0-5 3.6-5 6.8 0 0.5 0 1.1 0.1 1.6 -1.8 1.3-3.1 2.1-5.6 3.6 0-0.3 0.1-1.1 0.3-2.1 0.8-0.9 2-2.2 2-3.3 0-0.5-0.3-0.9-0.8-0.9 -1.4 0-2.4 2.1-2.7 3.5 -0.6 0.8-1.9 1.7-2.9 1.7 -0.8 0-1.1-0.8-1.2-1 2.6-0.9 5.9-4.5 5.9-7.8 0-0.7-0.3-2.2-2.4-2.2 -3.1 0-5.7 4.7-5.7 8.2 -1.1 0-1.5-1.2-1.5-2.1s0.4-1.8 0.4-2.1c0-0.3-0.1-0.6-0.5-0.6 -1 0-1.7 1.4-1.7 3 0.1 2.2 1.5 3.6 3.5 3.7 0.3 1.3 1.5 2.5 2.9 2.5 0.9 0 2-0.3 2.8-1 -0.1 0.5-0.1 0.9-0.2 1.3 -3 1.6-5.3 2.7-7.3 4.5 -1.6 1.5-2.5 3.4-2.5 4.9 0 2.1 1.3 4 4 4 3.2 0 5.6-2.5 6.8-6.1 0.6-1.7 0.8-4.1 0.9-6.3 3.2-1.8 4.7-2.8 6.4-4 0.2 0.4 0.4 0.6 0.7 0.8 -1.5 0.8-4.9 2.9-4.9 8 0 3.6 2.4 7.6 7.2 7.6 3.9 0 6.7-3.3 6.7-6.4C-241 400.8-242.6 398.2-245.7 398.2zM-264.7 407.9c-1 0-1.7-0.6-1.7-1.7 0-3 4.1-5.8 7.2-7.3C-259.8 403-261.2 407.7-264.7 407.9zM-262.4 391.5c0-2.3 2.3-6.6 3.6-6.2C-257.2 385.9-260.2 390.3-262.4 391.5zM-249.4 390.5c0-2.8 1.9-5.6 3-5.1C-245.2 386-247.3 388.7-249.4 390.5z"
        />
        <style>
          svg {fill: var(--px-ge-svg-logo-color, gray);}
        </style>
      </svg>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        Custom Title
      </div>
    </div>
  </div>
  <div
    className="c2"
  >
    <div
      className="c4"
    >
      Powered by React
    </div>
    <svg
      className="px-predix-svg-logo"
      height={10}
      viewBox="0 0 82.13 13.24"
    >
      <polygon
        className="cls-1"
        points="31.08 0 31.08 13.24 42.44 13.24 42.44 10.88 33.92 10.88 33.92 7.79 41.02 7.79 41.02 5.43 33.92 5.43 33.92 2.37 42.44 2.37 42.44 0 31.08 0"
      />
      <path
        className="cls-1"
        d="M49.3,10.84h2.17a4.93,4.93,0,0,0,3.64-1.32,4.12,4.12,0,0,0,1.11-3c-0.06-2-1.34-4.12-4.75-4.12H49.3v8.48Zm-2.84,2.4V0h5a8,8,0,0,1,5.72,2,6.4,6.4,0,0,1,1.87,4.48,6.35,6.35,0,0,1-1.78,4.62,7.91,7.91,0,0,1-5.81,2.11h-5Z"
      />
      <polygon
        className="cls-1 "
        points="62.57 13.24 65.41 13.24 65.41 0 62.57 0 62.57 6.57 62.57 13.24"
      />
      <path
        className="cls-1 "
        d="M6.2,0H0V13.24H2.86V9.09H6.2c3.83,0,5.57-2.41,5.57-4.67S10,0,6.2,0Zm0,6.72H2.86V2.37H6.2c1.88,0,2.74.92,2.74,2.05S8.08,6.72,6.2,6.72Z"
      />
      <path
        className="cls-1 "
        d="M27.09,13.24L23.56,8.81a4.57,4.57,0,0,0,3.5-4.39C27.06,2.16,25.32,0,21.49,0h-6.2V13.24h2.86V9.09H20.6l3.16,4.16h3.33ZM18.15,2.37h3.34c1.88,0,2.74.92,2.74,2.05s-0.86,2.3-2.74,2.3H18.15V2.37Z"
      />
      <polygon
        className="cls-1 "
        points="77.38 6.14 82.13 0 78.79 0 75.34 4.56 71.9 0 68.57 0 73.31 6.14 77.38 6.14"
      />
      <polygon
        className="cls-2 "
        points="73.31 7.08 68.55 13.24 71.88 13.24 75.34 8.67 78.8 13.24 82.13 13.24 77.37 7.08 73.31 7.08"
      />
      <style>
        .cls-1 { fill: #2886af; }.cls-2 { fill: #3ab4d4; }
      </style>
    </svg>
  </div>
</div>
`;

exports[`BrandingBar should render with hideLogo 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: inherit;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: var(--px-branding-bar-height,30px);
  max-height: 67px;
  min-height: 30px;
  background-color: var(--px-branding-bar-background-color,black);
  font-size: var(--px-branding-bar-font-size,15px);
  color: var(--px-branding-bar-logo-and-title-text-color,gray);
  padding: 0 1rem;
}

.c3 {
  margin-left: .5rem;
}

.c4 {
  font-size: 12px;
  margin-right: .3rem;
}

@media (max-width:500px) {
  .c0 {
    display: none !important;
    visibility: hidden;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        Custom Title
      </div>
    </div>
  </div>
  <div
    className="c2"
  >
    <div
      className="c4"
    >
      Powered by React
    </div>
    <svg
      className="px-predix-svg-logo"
      height={10}
      viewBox="0 0 82.13 13.24"
    >
      <polygon
        className="cls-1"
        points="31.08 0 31.08 13.24 42.44 13.24 42.44 10.88 33.92 10.88 33.92 7.79 41.02 7.79 41.02 5.43 33.92 5.43 33.92 2.37 42.44 2.37 42.44 0 31.08 0"
      />
      <path
        className="cls-1"
        d="M49.3,10.84h2.17a4.93,4.93,0,0,0,3.64-1.32,4.12,4.12,0,0,0,1.11-3c-0.06-2-1.34-4.12-4.75-4.12H49.3v8.48Zm-2.84,2.4V0h5a8,8,0,0,1,5.72,2,6.4,6.4,0,0,1,1.87,4.48,6.35,6.35,0,0,1-1.78,4.62,7.91,7.91,0,0,1-5.81,2.11h-5Z"
      />
      <polygon
        className="cls-1 "
        points="62.57 13.24 65.41 13.24 65.41 0 62.57 0 62.57 6.57 62.57 13.24"
      />
      <path
        className="cls-1 "
        d="M6.2,0H0V13.24H2.86V9.09H6.2c3.83,0,5.57-2.41,5.57-4.67S10,0,6.2,0Zm0,6.72H2.86V2.37H6.2c1.88,0,2.74.92,2.74,2.05S8.08,6.72,6.2,6.72Z"
      />
      <path
        className="cls-1 "
        d="M27.09,13.24L23.56,8.81a4.57,4.57,0,0,0,3.5-4.39C27.06,2.16,25.32,0,21.49,0h-6.2V13.24h2.86V9.09H20.6l3.16,4.16h3.33ZM18.15,2.37h3.34c1.88,0,2.74.92,2.74,2.05s-0.86,2.3-2.74,2.3H18.15V2.37Z"
      />
      <polygon
        className="cls-1 "
        points="77.38 6.14 82.13 0 78.79 0 75.34 4.56 71.9 0 68.57 0 73.31 6.14 77.38 6.14"
      />
      <polygon
        className="cls-2 "
        points="73.31 7.08 68.55 13.24 71.88 13.24 75.34 8.67 78.8 13.24 82.13 13.24 77.37 7.08 73.31 7.08"
      />
      <style>
        .cls-1 { fill: #2886af; }.cls-2 { fill: #3ab4d4; }
      </style>
    </svg>
  </div>
</div>
`;

exports[`BrandingBar should render with hidePowered 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: inherit;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: var(--px-branding-bar-height,30px);
  max-height: 67px;
  min-height: 30px;
  background-color: var(--px-branding-bar-background-color,black);
  font-size: var(--px-branding-bar-font-size,15px);
  color: var(--px-branding-bar-logo-and-title-text-color,gray);
  padding: 0 1rem;
}

.c3 {
  margin-left: .5rem;
}

@media (max-width:500px) {
  .c0 {
    display: none !important;
    visibility: hidden;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <svg
        height="20"
        viewBox="-280 372 50 50"
        width="20"
        x="0"
        y="0"
      >
        <path
          className="px-ge-svg-logo"
          d="M-255 422c-13.8 0-25-11.1-25-25 0-13.8 11.2-25 25-25s25 11.2 25 25C-230 410.9-241.2 422-255 422zM-255 373.2c-13.1 0-23.8 10.7-23.8 23.8s10.7 23.8 23.8 23.8 23.8-10.7 23.8-23.8S-241.8 373.2-255 373.2zM-233.8 404.1c0 0 0 0-0.1 0s-0.1 0-0.1-0.1c0 0 0.8-2.5 0.8-5.3 0-3-1.3-4.9-2.8-4.9 -1 0-1.7 0.7-1.7 1.7 0 1.9 2.3 2 2.3 6.1 0 1.7-0.4 3.3-0.9 5 -2.6 8.7-10.7 12.7-18.7 12.7 -3.7 0-6.3-0.8-7.1-1.1v-0.1l0.1-0.1c0.3 0.1 2.5 0.8 5.3 0.8 3 0 4.8-1.3 4.8-2.8 0-1-0.8-1.7-1.7-1.7 -1.9 0-2 2.4-6 2.4 -1.7 0-3.3-0.4-5.1-0.9 -8.6-2.6-12.7-10.7-12.7-18.8 0-3.9 1.1-7 1.1-7.1h0.1c0.1 0 0.1 0.1 0.1 0.1 -0.1 0.3-0.8 2.5-0.8 5.3 0 3 1.3 4.8 2.8 4.8 0.9 0 1.7-0.7 1.7-1.7 0-1.9-2.3-2.1-2.3-6.1 0-1.7 0.4-3.3 0.9-5.1 2.6-8.6 10.7-12.6 18.7-12.7 3.7 0 6.9 1.1 7.1 1.2v0.1c0 0.1-0.1 0.1-0.1 0.1s-2.1-0.9-5.3-0.9c-3 0-4.8 1.3-4.8 2.8 0 0.9 0.7 1.7 1.7 1.7 1.9 0 2-2.3 6-2.3 1.7 0 3.3 0.4 5.1 0.9 8.7 2.6 12.6 10.8 12.7 18.7C-232.6 401-233.8 404.1-233.8 404.1zM-245.7 398.2c-2.4 0-4.2 1.7-4.2 3.8 0 1.7 1 3.1 2.4 3.1 0.5 0 1-0.3 1-0.9 0-0.9-1.2-1.1-1.1-2.5 0-0.9 0.9-1.5 1.7-1.5 1.7 0 2.4 1.6 2.4 3.3 -0.1 2.6-2 4.3-4.2 4.3 -2.9 0-4.8-2.8-4.8-5.8 0-4.4 2.9-6.2 4.4-6.6 0 0 4 0.7 3.8-1 0-0.8-1.2-1.1-2-1.1 -0.9 0-1.8 0.3-1.8 0.3 -0.5-0.2-0.8-0.7-1-1.3 2.8-2.1 4.8-4.2 4.8-6.6 0-1.3-0.8-2.4-2.4-2.4 -2.8 0-5 3.6-5 6.8 0 0.5 0 1.1 0.1 1.6 -1.8 1.3-3.1 2.1-5.6 3.6 0-0.3 0.1-1.1 0.3-2.1 0.8-0.9 2-2.2 2-3.3 0-0.5-0.3-0.9-0.8-0.9 -1.4 0-2.4 2.1-2.7 3.5 -0.6 0.8-1.9 1.7-2.9 1.7 -0.8 0-1.1-0.8-1.2-1 2.6-0.9 5.9-4.5 5.9-7.8 0-0.7-0.3-2.2-2.4-2.2 -3.1 0-5.7 4.7-5.7 8.2 -1.1 0-1.5-1.2-1.5-2.1s0.4-1.8 0.4-2.1c0-0.3-0.1-0.6-0.5-0.6 -1 0-1.7 1.4-1.7 3 0.1 2.2 1.5 3.6 3.5 3.7 0.3 1.3 1.5 2.5 2.9 2.5 0.9 0 2-0.3 2.8-1 -0.1 0.5-0.1 0.9-0.2 1.3 -3 1.6-5.3 2.7-7.3 4.5 -1.6 1.5-2.5 3.4-2.5 4.9 0 2.1 1.3 4 4 4 3.2 0 5.6-2.5 6.8-6.1 0.6-1.7 0.8-4.1 0.9-6.3 3.2-1.8 4.7-2.8 6.4-4 0.2 0.4 0.4 0.6 0.7 0.8 -1.5 0.8-4.9 2.9-4.9 8 0 3.6 2.4 7.6 7.2 7.6 3.9 0 6.7-3.3 6.7-6.4C-241 400.8-242.6 398.2-245.7 398.2zM-264.7 407.9c-1 0-1.7-0.6-1.7-1.7 0-3 4.1-5.8 7.2-7.3C-259.8 403-261.2 407.7-264.7 407.9zM-262.4 391.5c0-2.3 2.3-6.6 3.6-6.2C-257.2 385.9-260.2 390.3-262.4 391.5zM-249.4 390.5c0-2.8 1.9-5.6 3-5.1C-245.2 386-247.3 388.7-249.4 390.5z"
        />
        <style>
          svg {fill: var(--px-ge-svg-logo-color, gray);}
        </style>
      </svg>
    </div>
    <div
      className="c2"
    >
      <div
        className="c3"
      >
        Custom Title
      </div>
    </div>
  </div>
</div>
`;
